
syntax = "proto3";

service AuthService {
  rpc validate_token (ValidateTokenRequest) returns (ValidateTokenResponse);
  rpc login (LoginRequest) returns (LoginResponse);
  rpc register(RegisterRequest) returns (RegisterResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string email = 3;
    string telegram_user_id = 4;
}

message RegisterResponse {
    string username = 1;
    string message = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
    string email = 3;
    string telegram_user_id = 4;
}

message LoginResponse {
    string token = 1;
    string expires_in = 2;
}

message ValidateTokenRequest {
    string token = 1;
}

message ValidateTokenResponse {
    bool is_valid = 1;
    string user_id = 2;
    string email = 3;
    string username = 4;
}

